_posts/2017-12-01-CocoaPods-1.4.0.markdown

```diff
@@ -0,0 +1,177 @@
+
+With CocoaPods 1.4.0 you are now able to specify a `script_phase` attribute within your podspec that executes as part of compilation of your pod, giving you full access to the Xcode build environment variables. Let's take a look at an example using our favorite sample pod 'CannonPodder':
+
+```ruby
+Pod::Spec.new do |s|
+  s.name         = 'CannonPodder'
+  s.version      = '1.0.0'
+  s.summary      = 'CannonPodder for the win.'
+  s.homepage     = 'http://cannonpodder-corp.local/cannonpodder-lib.html'
+  s.description  = 'All the Cannons'
+  s.source       = { :git => 'http://somewhere.com/cannonpodder.git', :tag => '1.0.0' }
+  s.license      = { :type => 'MIT', :text => 'Permission is hereby granted ...' }
+  
+  s.ios.deployment_target = '9.0'
+  s.osx.deployment_target = '10.10'
+
+  s.script_phase = { :name => 'Hello World', :script => 'echo "Hello World"' }
+  
+  s.source_files = 'Sources/**/*.{h,m,swift}'
+
+  s.private_header_files = 'Sources/Internal/**/*.h'
+end
+```
+
+The above will automatically add a script phase to the `CannonPodder-iOS` and `CannonPodder-macOS` targets generated by CocoaPods. It looks like this:
+
+<center><a href="http://feeds.cocoapods.org">
+{% breaking_image /assets/blog_img/CocoaPods-1.4.0/script_phases_podspec_1.png %}
+</a></center>
+
+If your pod supports multiple platforms, you can limit a script phase only to a specific platform, for example:
+
+```ruby
+s.ios.script_phase = { :name => 'Hello iOS World', :script => 'echo "Hello iOS World"' }
+```
+
+This will only add a script phase to the `CannonPodder-iOS` target.
+
+Additionally, there are times where you would like your script phases to be executed *before* compilation in order to perform a configuration step or a check. Support for this is included with CocoaPods 1.4.0 via the `:execution_position` attribute within each `script_phase` entry.
+
+```ruby
+s.script_phase = { 
+  :name => 'Hello World', 
+  :script => 'echo "Hello World"', 
+  :execution_position => :before_compile 
+}
+```
+
+Which generates the following:
+
+<center><a href="http://feeds.cocoapods.org">
+{% breaking_image /assets/blog_img/CocoaPods-1.4.0/script_phases_podspec_2.png %}
+</a></center>
+
+CocoaPods will automatically add the script phase to be executed before the `Compile Sources` phase executes. This effectively allows you to control at what stage you prefer a script phase to be executed.
+
+Here's a slightly more complicated example that demonstrates two script phases running at different stages:
+
+```ruby
+s.script_phases = [
+  { :name => 'Precompile', :script => './setup.sh', :execution_position => :before_compile },
+  { :name => 'Postcompile', :script => 'echo "yay!"', :execution_position => :after_compile }
+]
+```
+
+Currently, the available execution positions are `:before_compile`, `:after_compile` and `:any` which is the default.
```

Comment:
probably worth describing what `:any` does - at this point I'm not sure why I'd **not** define either `before` or `after`

Reply:
good point, I will update.